src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -83,26 +128,48 @@ const CharacterForm = ({ onClose }) => {
     species,
     numberOfFilms,
   }) => {
-    dispatch(
-      addCharacter({
-        id: nanoid(),
-        name,
-        eyeColor,
-        height,
-        gender: gender?.value,
-        birthYear,
-        homeworld: {
-          id: nanoid(),
-          name: homeworld,
-        },
-        species: {
+    const commonAttributes = {
+      name,
+      eyeColor,
+      height,
```

Comment:
Lets take this code to prepare the form submission data to the util file as well